termd-build-driver/src/main/java/org/jboss/pnc/termdbuilddriver/DefaultClientFactory.java

```diff
@@ -104,18 +108,23 @@ public BuildAgentClient createHttpBuildAgentClient(
             String executionId,
             Map<String, String> callbackHeaders) throws BuildAgentClientException {
 
+        List<Request.Header> headers = callbackHeaders.entrySet()
+                .stream()
+                .map(e -> new Request.Header(e.getKey(), e.getValue()))
+                .collect(Collectors.toList());
+
         HttpClientConfiguration configuration = null;
         try {
             URL callbackUrl = new URL(
                     StringUtils.stripEndingSlash(pncBaseUrl) + "/build-execution/" + executionId + "/completed");
-            Request callback = new Request("POST", callbackUrl, callbackHeaders);
+            Request callback = new Request(Request.Method.POST, callbackUrl.toURI(), headers);
             configuration = HttpClientConfiguration.newBuilder()
                     .termBaseUrl(terminalUrl)
                     .callback(callback)
                     .livenessResponseTimeout(30000L)
                     .retryConfig(retryConfig)
                     .build();
-        } catch (MalformedURLException e) {
+        } catch (MalformedURLException | URISyntaxException e) {
             new BuildAgentClientException("Invalid callback URL.", e);
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [RV: Exception created and dropped rather than thrown (RV_EXCEPTION_NOT_THROWN)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#rv-exception-created-and-dropped-rather-than-thrown-rv-exception-not-thrown).
This code creates an exception (or error) object, but doesn't do anything with it. For example, something like
It was probably the intent of the programmer to throw the created exception: